app/src/main/java/com/duckduckgo/app/brokensite/BrokenSiteViewModel.kt

```diff
@@ -16,79 +16,107 @@
 
 package com.duckduckgo.app.brokensite
 
+import android.net.Uri
+import androidx.annotation.VisibleForTesting
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import com.duckduckgo.app.brokensite.api.BrokenSiteSender
+import com.duckduckgo.app.brokensite.model.BrokenSite
+import com.duckduckgo.app.brokensite.model.BrokenSiteCategory
+import com.duckduckgo.app.brokensite.model.BrokenSiteCategory.*
 import com.duckduckgo.app.global.SingleLiveEvent
+import com.duckduckgo.app.global.absoluteString
+import com.duckduckgo.app.global.isMobileSite
 import com.duckduckgo.app.statistics.pixels.Pixel
 
-
 class BrokenSiteViewModel(private val pixel: Pixel, private val brokenSiteSender: BrokenSiteSender) : ViewModel() {
 
     data class ViewState(
-        val url: String? = null,
-        val message: String? = null,
+        val indexSelected: Int = -1,
+        val categorySelected: BrokenSiteCategory? = null,
         val submitAllowed: Boolean = false
     )
 
     sealed class Command {
-        object FocusUrl : Command()
-        object FocusMessage : Command()
         object ConfirmAndFinish : Command()
     }
 
     val viewState: MutableLiveData<ViewState> = MutableLiveData()
     val command: SingleLiveEvent<Command> = SingleLiveEvent()
-
+    var indexSelected = -1
+    val categories: List<BrokenSiteCategory> = listOf(
```

Comment:
any reason why this is not part of the `viewState`? It's being used in the `OnClickListener` of `categoriesSelection`

Reply:
The reason for not having it as part of the viewState is because is an intermediate step, when it changes the value the user has still not accepted the changes so it can still cancel the selection. I think the problem is the name of the variable, probably `preSelectedIndex` would make more sense?